tools/wavetool.py

```diff
@@ -0,0 +1,157 @@
+#! /usr/bin/python3
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2020 Intel Corporation. All rights reserved.
+
+import os
+import sys
+import argparse
+import numpy as np
+import scipy.signal as signal
+import scipy.io.wavfile as wavefile
+import scipy.fftpack as fftpack
+
+eps = 0.0000000001 # used to avoid zero division error
+
+def generateSinusoid(**params):
+    func = np.sin
+    if params['type'] == 'cosine':
+        func = np.cos
+    time = np.arange(0, float(params['duration']), 1.0/float(params['sample_rate']), dtype=np.float32)
+    data = float(params['amp']) * func(2 * np.pi * float(params['freq']) * time + float(params['phase']))
+    return np.reshape(np.repeat(data, int(params['channel'])),[len(data), int(params['channel'])])
+
+def parse_cmdline():
+    parser = argparse.ArgumentParser(add_help=True, formatter_class=argparse.RawTextHelpFormatter,
+        description='A Tool to Generate and Manipulate Wave Files.')
+    parser.add_argument('-v', '--version', action='version', version='%(prog)s 1.0')
+    parser.add_argument('-g', '--generate', type=str, help='generate specified types of wave\n'
+    'parameter format: "wave_type; amplitude(0.0~1.0); frequency; phase; sample rate; channel; format; duration"\n'
+    'Note: please use float point precision\n'
+    'Supported wave type: sine, cosine\n'
+    'Support format: S8, S16, S32, F32\n')
+    parser.add_argument('-o', '--output', type=str, help='Path to store generated files', default='.')
+    parser.add_argument('-p', '--preset', type=str, help='Commonly used wave presets\n'
+    'Supported presets are: sine_1k_16bit_48k_10s')
+    parser.add_argument('-c', '--compare', type=str, help='compare recorded wave and reference wave\n'
+    'parameter format: "mode;recorded_wave_path; reference_wave_path"\n'
+    'Supported comparison modes are: HWEchoRef')
+    return vars(parser.parse_args())
+
+def parse_params_and_gen_wav(cmd_param):
```

Comment:
Considering everything is in dictionaries, it should be easy to replace this single function with two names by two functions with a single name.